onnxruntime/core/providers/cpu/nn/tfidfvectorizer.cc

```diff
@@ -453,30 +453,36 @@ Status TfIdfVectorizer::ComputeImpl(OpKernelContext* ctx) const {
   } else if (input_dims.size() == 1) {
     b_dim = 1;
     C = input_dims[0];
-    if (C < 1) {
-      return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
-                    "Input shape must have either [C] or [B,C] dimensions where C > 0 and B > 0");
-    }
   } else if (input_dims.size() == 2) {
     B = input_dims[0];
     C = input_dims[1];
     b_dim = B;
-    if (B < 1 || C < 1) {
+    if (B < 1) {
       return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
-                    "Input shape must have either [C] or [B,C] dimensions where C > 0 and B > 0");
+                    "Input shape must have either [C] or [B,C] dimensions with B > 0.");
     }
   } else {
     return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
-                  "Input shape must have either [C] or [B,C] dimensions where C > 0 and B > 0");
+                  "Input shape must have either [C] or [B,C] dimensions with B > 0.");
   }
 
-  assert((b_dim * C) == total_items);
-
   // Frequency holder allocate [B..output_size_]
   // and init all to zero
   std::vector<uint32_t> frequencies;
   frequencies.resize(b_dim * impl.output_size_, 0);
 
+  if (input_shape.Size() == 0) {
+    // TfidfVectorizer may receive an empty input when it follows a Tokenizer
+    // (for example for a string containing only stopwords).
+    // The operator returns an output with null values of shape
```

Comment:
Not sure if my understanding is correct, so I try polishing this comment a bit. Feel free to ignore.
```suggestion
    // TfidfVectorizer returns a zero tensor of shape
```